src/Components/Rules.tsx

```diff
@@ -0,0 +1,64 @@
+import { useNavigate } from "react-router";
+import { useQuizContext } from "../Context/QuizProvider";
+import { quizzes } from "../data/getQuiz";
+
+export const Rules = () => {
+  const navigate = useNavigate();
+  const {
+    state: { quizName },
+  } = useQuizContext();
+  const quizDetail = quizzes.find((quiz) => quiz.quizName === quizName);
```

Comment:
You can understand my above statement here, You have not typed the variable quizDetail, yet it would not have shown error. If you hover over quizDetail, It automatically knows its type, Hope i make sense